spec/exoscale/requests/server_spec.rb

```diff
@@ -0,0 +1,92 @@
+# encoding: UTF-8
+
+require "minitest/autorun"
+require "fog/exoscale"
+
+describe Fog::Compute::Exoscale do
+  before do
+    @config = {
+      :exoscale_api_key           => ENV["EXO_API_KEY"],
+      :exoscale_secret_access_key => ENV["EXO_SECRET_KEY"]
+    }
+
+    @client = Fog::Compute::Exoscale.new(@config)
+  end
+  
+  describe "when creating a server with a new SG, new rules, new keypair" do
+    before do      
+      @ssh_key_pair         = @client.create_ssh_key_pair(:name => "fog_test_keypair_#{Random.new.rand(1..2000)}")['createsshkeypairresponse']['keypair']
+      
+      @security_group       = @client.create_security_group(:name => "fog_test_sg_#{Random.new.rand(1..2000)}")['createsecuritygroupresponse']['securitygroup']
+      
+      ingress_job_id        = @client.authorize_security_group_ingress(:securityGroupName => @security_group['name'],
+                                                                       :cidrList => '0.0.0.0/0', 
+                                                                       :startPort => '22', 
+                                                                       :endPort => '22', 
+                                                                       :protocol => 'TCP')['authorizesecuritygroupingressresponse']['jobid']
+      sleep 2 # because it's an async request
+      ingress_result        = @client.query_async_job_result(:jobId => ingress_job_id)['queryasyncjobresultresponse']
+      while (ingress_result['jobresult'] == nil)
+        sleep 2
+        ingress_result = @client.query_async_job_result(:jobId => ingress_job_id)['queryasyncjobresultresponse']
+      end
+      @ingress_rule_id      = ingress_result['jobresult']['securitygroup']['ingressrule'][0]['ruleid']
+      
+      egress_job_id         = @client.authorize_security_group_egress(:securityGroupName => @security_group['name'], 
+                                                                      :cidrList => '0.0.0.0/0',
+                                                                      :startPort => '22',
+                                                                      :endPort => '22',
+                                                                      :protocol => 'TCP')['authorizesecuritygroupegressresponse']['jobid']
+      sleep 2 # because it's an async request
```

Comment:
You should use `Fog.wait_for` here instead. Or in the code itself which would be way better i guess.